VideoLocker/src/main/java/org/edx/mobile/view/CourseDetailFragment.java

```diff
@@ -260,4 +282,120 @@ private ViewHolder createCourseDetailFieldViewHolder(LayoutInflater inflater, Li
+                    logger.debug("Enrollment successful:" + courseDetail.course_id);
+
+                    mEnrollButton.setText(R.string.view_course_button_text);
+                    Toast.makeText(getContext(), R.string.you_are_now_enrolled, Toast.LENGTH_SHORT).show();
+
+                    signalMyCourseListToRefresh();
+
+                    GetEnrolledCourseTask getEnrolledCourseTask =
+                            new GetEnrolledCourseTask(getActivity(), courseDetail.course_id) {
+                                @Override
+                                public void onSuccess(EnrolledCoursesResponse course) {
+                                    if (course != null) {
+                                        environment.getRouter().showCourseDashboardTabs(getActivity(), environment.getConfig(), course, false);
+                                    } else {
+                                        Toast.makeText(getContext(), R.string.cannot_show_dashboard, Toast.LENGTH_SHORT).show();
+                                    }
+                                }
+
+                                @Override
+                                public void onException(Exception ex) {
+                                    logger.error(ex);
+                                    Toast.makeText(getContext(), R.string.enrollment_failure, Toast.LENGTH_SHORT).show();
+                                }
+                            };
+                    getEnrolledCourseTask.execute();
+
+                } else {
+                    logger.debug("Enrollment failure:" + courseDetail.course_id);
+                    Toast.makeText(getContext(), R.string.enrollment_failure, Toast.LENGTH_SHORT).show();
+                }
+            }
+
+            @Override
+            public void onException(Exception ex) {
+                logger.error(ex);
+                Toast.makeText(getContext(), R.string.enrollment_failure, Toast.LENGTH_SHORT).show();
+            }
+        };
+        enrollForCourseTask.execute();
+    }
+
+    /**
+     * Open course dashboard for given course from the enrollments list cache.
+     */
+    private void openCourseDashboard() {
+        ServiceManager api = environment.getServiceManager();
+        try {
+            List<EnrolledCoursesResponse> enrolledCoursesResponse = api.getEnrolledCourses(true);
+            for (EnrolledCoursesResponse course : enrolledCoursesResponse) {
+                if (course.getCourse().getId().equals(courseDetail.course_id)) {
+                    environment.getRouter().showCourseDashboardTabs(getActivity(), environment.getConfig(), course, false);
+                }
+            }
+        } catch (Exception exception) {
+            logger.debug(exception.toString());
+            Toast.makeText(getContext(), R.string.cannot_show_dashboard, Toast.LENGTH_SHORT).show();
+        }
+    }
+    
+    private void signalMyCourseListToRefresh() {
+        Intent intent = new Intent(RELOAD_MY_COURSES);
```

Comment:
The event bus would be a great fit for this instead of using the broadcast manager:

1) Create an event such as `EnrolledInCourseEvent`
2) Inside `EnrollForCourseTask.call`, when successful, post an event `EventBus.getDefault().post(new EnrolledInCourseEvent())`
3) In the "My Courses" fragment, listen for the event: `public void onEventMainThread(EnrolledInCourseEvent event) { refresh(); }`